api/src/main/java/org/openmrs/module/pihcore/search/PihLuceneAnalyzerFactory.java

```diff
@@ -0,0 +1,43 @@
+package org.openmrs.module.pihcore.search;
+
+import org.apache.lucene.analysis.core.KeywordTokenizerFactory;
+import org.apache.lucene.analysis.core.LowerCaseFilterFactory;
+import org.apache.lucene.analysis.core.WhitespaceTokenizerFactory;
+import org.apache.lucene.analysis.miscellaneous.ASCIIFoldingFilterFactory;
+import org.apache.lucene.analysis.ngram.EdgeNGramFilterFactory;
+import org.apache.lucene.analysis.ngram.NGramFilterFactory;
+import org.apache.lucene.analysis.standard.ClassicFilterFactory;
+import org.hibernate.search.annotations.Factory;
+import org.hibernate.search.cfg.SearchMapping;
+import org.openmrs.api.db.hibernate.search.LuceneAnalyzers;
+
+
+public class PihLuceneAnalyzerFactory {
+    @Factory
+    public SearchMapping getSearchMapping() {
+        SearchMapping mapping = new SearchMapping();
```

Comment:
One approach would be for this SearchMapping object to be defined as a Spring bean, where there is a default bean supplied by core, and modules can override this by adding their own bean that implements the same interface and the code within here chooses one vs. the other based on some configured sort order.   There are examples of doing this within the reporting module and elsewhere that I could point you to.  Not sure if that is the most efficient model or not, but it's an option.

Reply:
@mseaton @brandones this seems like a good approach to me!